Updates to zend-expressive-router 2.4.0

weierophinney opened the pull request:
Only change in that release is that the `Route` instance now triggers a deprecation notice if non-MiddlewareInterface `$middleware` is provided at instantiation. This patch updates the tests to mock the `MiddlewareInterface` and provide it as an instance whenever a new `Route` instance is created. References to the `'middleware'` default in route specifications also reference it.

weierophinney committed:
Updates to zend-expressive-router 2.4.0

Only change in that release is that the `Route` instance now triggers a
deprecation notice if non-MiddlewareInterface `$middleware` is provided
at instantiation. This patch updates the tests to mock the
`MiddlewareInterface` and provide it as an instance whenever a new
`Route` instance is created. References to the `'middleware'` default in
route specifications also reference it.

michalbundyra reviewed (commented):
Needs update of LEGACY_DEPS in Travis CI config. See build and error:

```
  Problem 1
    - symfony/console v4.0.6 requires php ^7.1.3 -> your PHP version (5.6.32) does not satisfy that requirement.
    - symfony/console v4.0.6 requires php ^7.1.3 -> your PHP version (5.6.32) does not satisfy that requirement.
    - symfony/console v4.0.6 requires php ^7.1.3 -> your PHP version (5.6.32) does not satisfy that requirement.
    - Installation request for symfony/console (locked at v4.0.6) -> satisfiable by symfony/console[v4.0.6].
```

weierophinney committed:
Adds more legacy deps configuration for travis

- symfony/console and symfony/finder need older releases.
- `--no-scripts` as a default composer flag.

michalbundyra reviewed (approved):
LGTM 👍
